build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1997,6 +2018,11 @@ message CacheCapabilities {
   // [BatchUpdateBlobs][build.bazel.remote.execution.v2.ContentAddressableStorage.BatchUpdateBlobs]
   // requests.
   repeated Compressor.Value supported_batch_update_compressors = 7;
+
+  // The digest of the
```

Comment:
Giving this part some more thought: you may also need to extend GetCapabilitiesRequest to contain a list of supported digest functions. Otherwise this field may return a digest that is incompatible with the client. There need to be some negotiation here.

Reply:
Then it might need to be a combination of Digest and Compressor. Because the dictionary registry blob could also be compressed with a non-dictionary compressor right? 🤔

Reply:
There is no need to account for that, as it's always possible to fetch blobs with the IDENTITY compressor. I think that most clients would actually do that when fetching a dictionary. With digest functions it's a different story, as a client may not actually be able to construct a bytestream URL if it does not provide support for a given digest function.

Reply:
If Digest functions are unavailable in GetCapRequest, we should assume that it's SHA256 right?

Reply:
No. There is nothing in the Remote Execution protocol right now that gives SHA-256 a special status over the others.

Reply:
On one hand, I do agree that some negotiation seems to be needed here for the client/server to agree on the digest. On the other hand, I am a bit afraid of the scope creep of having to extend GetCapabilitiesRequest. For example: if the client were to only support `SHA256` digest function, should the server still return other digest functions in CacheCapabilities? What if the server does not support SHA256? should it just err out?

I guess we could just left it undecided in the spec?

Reply:
> On the other hand, I am a bit afraid of the scope creep of having to extend GetCapabilitiesRequest. For example: if the client were to only support SHA256 digest function, should the server still return other digest functions in CacheCapabilities?

It still may. We're not going to adjust anything there. The digest functions would only be used to control the format of digests returned as part of GetCapabilities(). Not any of the other fields.

Another option would be to simply add a dedicated call for negotiating compression dictionaries, as opposed to having it part of GetCapabilities().

> What if the server does not support SHA256? should it just err out?

You mean if the client only announces support for SHA-256 and the server does not support it? Then it has no way to return the dictionary in that format. So in that case it may return a capabilities response that does not announce the support for ZSTD_DICT.